### PR TITLE
CB-11835 Remove Mock IDBroker Mappings from API test project and E2E tests

### PR DESCRIPTION
**Pull request by aszegedi**
This is a tech debt task. Mock IDBroker has caused multiple production down issues. We shall remove it from CB, based on the [EPIC CB-11728](https://jira.cloudera.com/browse/CB-11728)

### CODE REVIEW

File: integration-test/src/main/resources/application.yml [hunk 1]
```diff
@@ -333,6 +342,8 @@ integrationtest:
         instanceProfile: "arn:aws:iam::1234567890:instance-profile/mock.testing.instance.profile"
       baseLocation: "s3a://mock-test"
       fileSystemType: S3
+      dataAccessRole:
```
Comment (doktoric) [inline integration-test/src/main/resources/application.yml:345]: we already added this to e2e tests on jenkins right @aszegedi ?

Comment (aszegedi) [reply integration-test/src/main/resources/application.yml:345]: What we really-really need here is the:
```
idbmms:
    # Here we need a valid idbroker mapping management host (Manowar Dev)!
    host:
```

**Comment by aszegedi** [comment]
I've checked related `CDP CLI` commands on `local environment`:
- `cdp environments set-id-broker-mappings`
- `cdp environments get-id-broker-mappings`
- `cdp environments sync-id-broker-mappings`
- `cdp environments get-id-broker-mappings-cli-for-env`
- `cdp environments get-id-broker-mappings-sync-status`
- `cdp environments delete-id-broker-mappings`

all worked fine.

File: idbmms-connector/src/main/java/com/sequenceiq/cloudbreak/idbmms/config/IdbmmsClientConfig.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package com.sequenceiq.cloudbreak.idbmms.config;
+
+import org.springframework.context.annotation.Configuration;
+
+import io.netty.util.internal.StringUtil;
+
+@Configuration
+public class IdbmmsClientConfig {
+
+    private final String callingServiceName;
```
Comment (biharitomi) [inline idbmms-connector/src/main/java/com/sequenceiq/cloudbreak/idbmms/config/IdbmmsClientConfig.java:10]: Shouldn't this be a property value like the following?
```
@Value("${altus.ums.caller:cloudbreak}")

**Review by biharitomi — APPROVED** [review]

File: idbmms-connector/src/main/java/com/sequenceiq/cloudbreak/idbmms/config/IdbmmsClientConfig.java (hunk 2 above)
Comment (aszegedi) [reply idbmms-connector/src/main/java/com/sequenceiq/cloudbreak/idbmms/config/IdbmmsClientConfig.java:12]: I've got it fixed, based on suggestion:
```
@Autowired
public IdbmmsClientConfig(@Value("${altus.idbmms.caller:cloudbreak}") String callingServiceName) {
   this.callingServiceName = callingServiceName;
}
```
Thx, @biharitomi

**Review by biharitomi — APPROVED** [review]

**Review by foldik — APPROVED** [review]
LGTM

File: idbmms-connector/build.gradle [hunk 3]
```diff
@@ -24,6 +24,7 @@ dependencyManagement {
 
 dependencies {
   implementation project(':grpc-common')
+  implementation project(':auth-connector')
```
Comment (lnardai) [inline idbmms-connector/build.gradle:27]: Why is this required here? Client should not rely on the auth-connector.
Maybe you wanted auth common if you use some Crn magic?

**Review by lnardai — CHANGES_REQUESTED** [review]
Please remove auth-connector dependency from idbroker-connector, they should not depend on each other.
Evaluate the baseclass extension idea.

File: idbmms-connector/build.gradle (hunk 3 above)
Comment (aszegedi) [reply idbmms-connector/build.gradle:27]: This is needed for `String requestId = RequestIdUtil.getOrGenerate(MDCUtils.getRequestId());`

Comment (lnardai) [reply idbmms-connector/build.gradle:27]: You could use this version of to generate a RequestId:
https://github.com/hortonworks/cloudbreak/blob/1fbceb9382a26df2277e1d6f2df2839f46f48e6f/common/src/main/java/com/sequenceiq/cloudbreak/logger/MDCBuilder.java#L69-L69

**Review by lnardai — APPROVED** [review]
